### PR TITLE
Get config fix

### PR DESCRIPTION
**Pull request by DrRayWang**
Fixed the [issue 294](https://github.com/bloomberg/chef-bach/issues/294), and made related code change in code case to apply get_config!().

### CODE REVIEW

**Comment by amithkanand** [comment]
Recipe changes?

**Comment by DrRayWang** [comment]
Yes all the recipes which use old get_config to access chef vault item are changed to using new get_config!()

**Comment by http-418** [comment]
This failed to build for me.  Couldn't chef nodes:
https://gist.github.com/http-418/82bebe00669100128c58

**Comment by DrRayWang** [comment]
Andrew, I tested it again, I didn't get your errors. Could you check your test environment setting, and try it again.

**Comment by pu239ppy** [comment]
Worked for me, however you should remove FLUSH PRIVILEGES it is unnecesary, since you are using GRANT and not directly modifying privilege tables by INSERTing or DELETEing

**Comment by DrRayWang** [comment]
we can work on "remove FLUSH PRIVILEGES" in a different PR.

**Comment by kiiranh** [comment]
You could save a code-test-review cycle if you make the change in this PR!

File: cookbooks/bcpc/libraries/utils.rb [hunk 1]
```diff
@@ -115,8 +115,8 @@ def delete_config(key)
   end
 end
 
-def get_config!(key)
-  value = get_config(key)
+def get_config!(key,item=node.chef_environment,bag="configs")
+  value = get_config(key,item,bag)
   raise "Failed to find value for #{key}!" if value.nil?
```
Comment (kiiranh) [inline cookbooks/bcpc/libraries/utils.rb]: It would have been good to print `item` along with `key` in the Error message

**Comment by pu239ppy** [comment]
Built a cluster with this pull request tested with C-A-R

**Pull request merged**